crates/miden-lib/src/account/faucets/mod.rs

```diff
@@ -278,14 +278,16 @@ pub fn create_basic_fungible_faucet(
 ) -> Result<(Account, Word), FungibleFaucetError> {
     let distribute_proc_root = BasicFungibleFaucet::distribute_digest();
 
-    let auth_component: AuthRpoFalcon512Acl = match auth_scheme {
+    let auth_component: AccountComponent = match auth_scheme {
         AuthScheme::RpoFalcon512 { pub_key } => AuthRpoFalcon512Acl::new(
             pub_key,
             AuthRpoFalcon512AclConfig::new()
                 .with_auth_trigger_procedures(vec![distribute_proc_root])
                 .with_allow_unauthorized_input_notes(true),
         )
-        .map_err(FungibleFaucetError::AccountError)?,
+        .map_err(FungibleFaucetError::AccountError)?
+        .into(),
+        AuthScheme::NoAuth => NoAuth::new().into(),
```

Comment:
Let's create an issue (unless we already have one) to remove this option (i.e., we should not be able to create a basic fungible faucet with `NoAuth` component).